### PR TITLE
ddl: fix pre-split region timeout constraint not work when create table

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

close https://github.com/pingcap/tidb/issues/17458

As the title said.

### What is changed and how it works?

Use `context.WithTimeout`.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test 

Side effects

- No

### Release note 

- Fix pre-split region timeout constraint not work when create table.

### CODE REVIEW

**Comment by wjhuang2016** [comment]
Need to cherry-pick?

File: store/tikv/split_region.go [hunk 1]
```diff
@@ -207,18 +207,19 @@ func (s *tikvStore) SplitRegions(ctx context.Context, splitKeys [][]byte, scatte
...
-		}
-	})
-
+func (s *tikvStore) scatterRegion(ctx context.Context, regionID uint64) error {
 	logutil.BgLogger().Info("start scatter region",
 		zap.Uint64("regionID", regionID))
-	bo := NewBackoffer(context.Background(), scatterRegionBackoff)
+	bo := NewBackoffer(ctx, scatterRegionBackoff)
 	for {
 		err := s.pdClient.ScatterRegion(context.Background(), regionID)
```
Comment (zimulala) [inline store/tikv/split_region.go:215]: Do we need to update this `context`?

File: ddl/split_region.go [hunk 2]
```diff
@@ -18,34 +18,39 @@ import (
...
+		regionIDs = append(regionIDs, splitRecordRegion(ctxWithTimeout, store, def.ID, scatter))
 	}
 	if scatter {
-		waitScatterRegionFinish(store, regionIDs...)
+		waitScatterRegionFinish(ctxWithTimeout, store, regionIDs...)
 	}
 }
 
-func splitTableRegion(store kv.SplittableStore, tbInfo *model.TableInfo, scatter bool) {
+func splitTableRegion(ctx sessionctx.Context, store kv.SplittableStore, tbInfo *model.TableInfo, scatter bool) {
```
Comment (zimulala) [inline ddl/split_region.go:40]: It seems we don't use `ctx`, so why add this argument?

File: ddl/split_region.go [hunk 3]
```diff
@@ -118,11 +123,17 @@ func splitIndexRegion(store kv.SplittableStore, tblInfo *model.TableInfo, scatte
 	return regionIDs
 }
 
-func waitScatterRegionFinish(store kv.SplittableStore, regionIDs ...uint64) {
+func waitScatterRegionFinish(ctx context.Context, store kv.SplittableStore, regionIDs ...uint64) {
 	for _, regionID := range regionIDs {
 		err := store.WaitScatterRegionFinish(regionID, 0)
```
Comment (zimulala) [inline ddl/split_region.go:128]: Could we add the argument of `ctx` to `WaitScatterRegionFinish`? Then we can remove line132-137.

Comment (crazycs520) [reply ddl/split_region.go]: Great, done

File: store/tikv/split_region.go (hunk 1 above)
Comment (crazycs520) [reply store/tikv/split_region.go]: Great, done

File: ddl/split_region.go (hunk 2 above)
Comment (crazycs520) [reply ddl/split_region.go:39]: Need `ctx` to get sessionVars.

**Pull request merged**